compiler/rustc_middle/src/mir/statement.rs

```diff
@@ -958,5 +969,5 @@ impl RawPtrKind {
 
 #[derive(Clone, TyEncodable, TyDecodable, HashStable, TypeFoldable, TypeVisitable)]
 pub enum StmtDebugInfo<'tcx> {
-    AssignRef(Local, Place<'tcx>),
+    AssignRef(Local, Option<Place<'tcx>>),
```

Comment:
I'm not convinced by the benefit of allowing `None` here. Do you mind splitting this change from other tweaks to DSE, so we can see what it does in isolation?